### ISSUE TITLE
Continuation_already_taken crashes toplevel

### ISSUE THREAD

**Issue by progman1** [ocamllabs/ocaml-multicore#479]
an early effort at the tutorial's iterator-to-generator exercise:
```
[@@@warning "-5-26"]

type ('a, 'container) iterator = ('a -> unit) -> 'container -> unit
type 'a generator = unit -> 'a option

type ('a,'container) iter2gen =
  ('a, 'container) iterator     (* List.iter *)
  -> 'container
  -> 'a generator

effect Hold: unit
let iter2gen : _ iter2gen = fun iter c ->
  let r = ref None in
  let suspending_f x =
    r:=Some x;
    (perform Hold : unit)
  in
  let next =
    match iter suspending_f c with
    | () -> fun ()->None
    | effect Hold k ->
       fun()->
         let x = !r in
         continue k ();
         x

   in
   fun()-> next()


let _=
  let gen = iter2gen List.iter in
  let gen =gen [1;2;3] in
  let gen() =match gen() with None->"?" | Some x-> string_of_int x in
  Printf.printf "%s\n" (gen());
  Printf.printf "%s\n" (gen());
```

segfaults the toplevel.
fortunately the compiled version showed me the error of my ways.

4.10.0+multicore on x86-64

**Comment by ctk21** [comment]
I've managed to recreate this; it is affecting the `parallel_minor_gc` branch and not the newer `4.12+domains+effects` branch.

**Comment by kayceesrk** [comment]
If we can confirm that the error does not occur in `4.12+domains+effects`, can we close this issue?

**Comment by ctk21** [comment]
Yes, I've tested that this error does not occur with `4.12+domains+effects`. 

I plan to cherry-pick the addition of the testcase to `4.12+domains+effects` when we merge #509.

**Comment by sadiqj** [comment]
Closed with the merging of #509
